### PR TITLE
feat(api): add hatchery consumer

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/hatchery/vsphere/hatchery.go [hunk 1]
```diff
@@ -42,6 +42,7 @@ func (h *HatcheryVSphere) Init(config interface{}) (cdsclient.ServiceConfig, err
 	}
 	cfg.Host = sConfig.API.HTTP.URL
 	cfg.Token = sConfig.API.Token
+	cfg.V2Token = sConfig.API.Token
```
Comment (yesnault) [inline engine/hatchery/vsphere/hatchery.go:45]:
```suggestion
	cfg.TokenV2 = sConfig.API.Token
```

**Review by yesnault — CHANGES_REQUESTED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18158.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✘

File: engine/api/api_routes.go [hunk 2]
```diff
@@ -46,6 +46,7 @@ func (api *API) InitRouter() {
 	r.Handle("/auth/consumer/local/askReset", ScopeNone(), r.POST(api.postAuthLocalAskResetHandler, service.OverrideAuth(service.NoAuthMiddleware), MaintenanceAware()))
 	r.Handle("/auth/consumer/local/reset", ScopeNone(), r.POST(api.postAuthLocalResetHandler, service.OverrideAuth(service.NoAuthMiddleware), MaintenanceAware()))
 	r.Handle("/auth/consumer/builtin/signin", ScopeNone(), r.POST(api.postAuthBuiltinSigninHandler, service.OverrideAuth(service.NoAuthMiddleware), MaintenanceAware()))
+	r.Handle("/auth/consumer/hatchery/signin", ScopeNone(), r.POSTv2(api.postAuthHatcherySigninHandler, service.OverrideAuth(service.NoAuthMiddleware), MaintenanceAware()))
```
Comment (richardlt) [inline engine/api/api_routes.go:49]: Not in v2 ?

**Review by yesnault — APPROVED** [review]

File: engine/api/authentication/consumer.go [hunk 3]
```diff
@@ -8,13 +8,30 @@ import (
...
+	return &c, nil
+}
+
 func NewConsumerWorker(ctx context.Context, db gorpmapper.SqlExecutorWithTx, name string, hatcheryConsumer *sdk.AuthUserConsumer) (*sdk.AuthUserConsumer, error) {
 	c := sdk.AuthUserConsumer{
 		AuthConsumer: sdk.AuthConsumer{
-			Name:            name,
-			ParentID:        &hatcheryConsumer.ID,
-			Type:            sdk.ConsumerBuiltin,
-			ValidityPeriods: sdk.NewAuthConsumerValidityPeriod(time.Now(), 24*time.Hour),
```
Comment (richardlt) [inline engine/api/authentication/consumer.go:17]: Why do we remove this validity period ?

File: engine/api/authentication/dao_hatchery_consumer.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	if err != nil {
+		return nil, err
+	}
+	isValid, err := gorpmapping.CheckSignature(dbAuthConsumerHatchery, dbAuthConsumerHatchery.Signature)
+	if err != nil {
+		return nil, err
+	}
+	if !isValid {
+		log.Error(ctx, "authentication.getAllAuthConsumerHatcheries> auth consumer hatchery %s data corrupted", dbAuthConsumerHatchery.ID)
+		return nil, err
```
Comment (richardlt) [inline engine/api/authentication/dao_hatchery_consumer.go:65]: err is nil here but should not, do not log.Error but return the error

File: engine/api/authentication/hatchery/hatchery.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+	"context"
+	"time"
+
+	"github.com/rockbears/log"
+
+	"github.com/ovh/cds/engine/api/authentication"
+	"github.com/ovh/cds/sdk"
+)
+
+var SessionDuration = 365 * 24 * time.Hour
```
Comment (richardlt) [inline engine/api/authentication/hatchery/hatchery.go:13]: Currently 1 h, we should not keep the session for 1 year

File: engine/hatchery/serve.go [hunk 6]
```diff
@@ -115,6 +116,77 @@ func (c *Common) GetPrivateKey() *rsa.PrivateKey {
...
+	ticker := time.NewTicker(10 * time.Second)
+	defer ticker.Stop()
+
+	serviceConfig, err := service.ParseServiceConfig(srvConfig)
+	if err != nil {
+		return err
+	}
+
+	var pubKey []byte
+	if c.PrivateKey != nil {
```
Comment (richardlt) [inline engine/hatchery/serve.go:135]: Should never be nil

**Review by richardlt — CHANGES_REQUESTED** [review]

**Review by richardlt — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18163.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18164.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#18166.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Pull request merged**